packages/util/src/extractTime.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017-2020 @polkadot/util authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Time } from './types';
+
+const HRS = 60 * 60;
+const DAY = HRS * 24;
+
+/**
+ * @name addTime
+ * @summary Add together two Time arrays
+ */
+
+function addTime (a: Time, b: Time): Time {
+  return {
+    days: a.days + b.days,
+    hours: a.hours + b.hours,
+    minutes: a.minutes + b.minutes,
+    seconds: a.seconds + b.seconds
+  };
+}
+
+const ZERO = { days: 0, hours: 0, minutes: 0, seconds: 0 };
```

Comment:
Re-looking at this - this is actually quite "weird" since it doesn't have ms. JS dates are to the ms. (I understand this solves a specific problem, however we need to align - if the caller doesn't have ms, he needs to add them)